text/0218-standard-telemetry-query-api.md

```diff
@@ -0,0 +1,130 @@
+
+  ```protobuf
+  message Span {
+    string trace_id = 1;
+    string parent_id = 2;
+    string id = 3;
+    string kind = 4;
+    string name = 5;
+    fixed64 timestamp = 6;
+    uint64 duration = 7;
+    Endpoint local_endpoint = 8;
+    Endpoint remote_endpoint = 9;
+    repeated Annotation annotations = 10;
+    map<string, string> tags = 11;
+    bool debug = 12;
+    bool shared = 13;
+  }
+
+  // Demonstration of a Trace object as described in the API referance
+  message Trace {
+    repeated Span = 1;
+  }
+  ```
+
+- [Jaeger Span and Trace Objects](https://github.com/jaegertracing/jaeger-idl/blob/main/proto/api_v2/model.proto)
+
+  ```protobuf
+  message Span {
+    bytes trace_id = 1;
+    bytes span_id = 2;
+    string operation_name = 3;
+    repeated SpanRef references = 4;
+    uint32 flags = 5;
+    google.protobuf.Timestamp start_time = 6;
+    google.protobuf.Duration duration = 7;
+    repeated KeyValue tags = 8;
+    repeated Log logs = 9;
+    Process process = 10;
+    string process_id = 11;
+    repeated string warnings = 12;
+  }
+
+  message Trace {
+    message ProcessMapping {
+        string process_id = 1;
+        Process process = 2;
+    }
+    repeated Span spans = 1;
+    repeated ProcessMapping process_map = 2;
+    repeated string warnings = 3;
+  }
+  ```
+
+- [Tempo Span and Trace Objects](https://grafana.com/docs/tempo/latest/api_docs/) - From [Tempo docs](https://grafana.com/docs/tempo/latest/api_docs/#query) it is not clear what is exactly the structure of a Trace, though we can see a [Go implementation of a proto definition](https://github.com/grafana/tempo/blob/main/pkg/tempopb/trace/v1/trace.pb.go#L307).
+
+From looking at these different, opinionated Span and Trace objects we can immediately see that the burden for mapping these objects to a common format falls on the consumer side.
+
+## Trade-offs and mitigations
+
+### Standard API vs. Backend Proxy Library
+
+- In order to make the shift toward this standard more pleasant from the telemetry consumers side, while reducing the dependency on external platforms, a dedicated contrib repo should be created with a proxy client that will abstract the details of specific implementations (Jaeger, Tempo, etc.) so that the user will be able to switch one observability backend to another while enabling telemetry consumers to query any supported observability backend in a common format.
+
+## Prior art and alternatives
+
+- The proxy library mentioned above can be inspired by this [POC](https://github.com/lucasponce/jaeger-proto-client) made by @lucasponce that can plug any Jaeger solution.
+- [Tempo TraceQL](https://github.com/grafana/tempo/blob/main/docs/design-proposals/2022-04%20TraceQL%20Concepts.md), a language for selecting traces that Tempo will implement. This language is currently only focused on trace selection. This is an example for a specific implementation for querying syntax.
+
+## Open questions
```

Comment:
I'd add that the actual mandatory and optional API capabilities is an open question - @itaykat I'm sure you have an initial idea of what capabilities you would like to see in the API. What is the process of making the next step in this discussion?

Reply:
Please see https://github.com/open-telemetry/oteps/pull/218#issuecomment-1247114097